express/src/controllers/sign-in.ts

```diff
@@ -116,6 +116,16 @@ export const forgotPasswordForm = async function (req: Request, res: Response) {
 };
 
 export const checkEmailPasswordReset = async function (req: Request, res: Response) {
+    await forgotPassword(req, res);
+    res.render("check-email-password-reset.njk");
+};
+
+export const resendForgotPassword = async function (req: Request, res: Response) {
+    await forgotPassword(req, res);
```

Comment:
Does this behave properly if the password is an empty string?  I feel like this will generate one of the can't render twice type errors.

Reply:
@JohnWattsGDS Shouldn't the validation of email be done via middleware in the route?

Reply:
This would solve the problem. `middleware/validators/emailIsPresentInSession` can be inserted into the middleware chaing.

Reply:
Done.